static/scenes/railroad/js/game.js

```diff
@@ -54,11 +91,32 @@ export class Game {
 
   /**
    * Handles the main logic for the game by making each system update.
-   *
-   * TODO: There should be some delta time calculated here and passed to the
-   * update methods so the game runs at a consistent time rate.
    */
   update() {
-    this.cameraSystem.update();
+    const nowSeconds = Date.now() / 1000;
+    const deltaSeconds = nowSeconds - this.previousSeconds;
+    this.cameraSystem.update(deltaSeconds);
+    this.scoreboard.onFrame(deltaSeconds);
+    this.previousSeconds = nowSeconds;
+    
+  }
+
+  setUpListeners() {
+    this.clickListener = this.renderer.domElement.addEventListener('click', (click) => {
+      this.handleClick(click);
+    });
+  }
+
+  handleClick(clickEvent) {
+    // calculate pointer position in normalized device coordinates
+    // (-1 to +1) for both components
+    const ray = new THREE.Vector2(0, 0);
+    ray.x = (clickEvent.clientX / this.renderer.domElement.width) * 2 - 1;
+    ray.y = -(clickEvent.clientY / this.renderer.domElement.height) * 2 + 1; 
+    this.raycaster.setFromCamera(ray, this.camera);
+    const intersects = this.raycaster.intersectObjects(this.placeholderScene.scene.children, true);
+    this.scoreboard.addScore(intersects.length);
```

Comment:
This would be better put into a class similar to camera-system, so we don't have game logic directly in this class.

Reply:
Done, thanks for the feedback. I might not have extracted the right part or set up the api right. I'm open to further feedback.